model/messages/collection.go

```diff
@@ -42,17 +44,24 @@ type UntrustedClusterBlock struct {
 
 // ToInternal returns the internal representation of the type.
 func (ub *UntrustedClusterBlock) ToInternal() *cluster.Block {
-	block := &cluster.Block{
+	txs := make([]*flow.TransactionBody, 0, len(ub.Payload.Collection))
+
+	for _, tx := range ub.Payload.Collection {
+		txs = append(txs, &tx)
+	}
+
+	collection, err := flow.NewCollection(flow.UntrustedCollection{Transactions: txs})
+	if err != nil {
+		panic(fmt.Sprintf("invalid untrusted input in ToInternal: %v", err))
```

Comment:
We should return an error here instead of panicking. A large part of the purpose of `UntrustedClusterBlock` is to remove a surface area for attack (nil-able fields which might not be validated later on and cause a nil pointer crash). If we panic when validation fails in `NewCollection`, we're losing most of the benefit of performing the validation here.